platform/android/MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/module/telemetry/SchemaTest.java

```diff
@@ -0,0 +1,261 @@
+package com.mapbox.mapboxsdk.module.telemetry;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.annotations.SerializedName;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class SchemaTest {
+  private static final String MAP_CLICK = "map.click";
+  private static final String MAP_DRAG = "map.dragend";
+  private static final String MAP_LOAD = "map.load";
+  private static ArrayList<JsonObject> schemaArray;
+
+  @BeforeClass
+  public static void downloadSchema() throws Exception {
+    unpackSchemas();
+  }
+
+  @Test
+  public void checkMapClickEventSize() throws Exception {
+    JsonObject schema = grabSchema(MAP_CLICK);
+    List<Field> fields = grabClassFields(MapClickEvent.class);
+
+    assertEquals(schema.size(), fields.size());
+  }
+
+  @Test
+  public void checkMapClickEventFields() throws Exception {
+    JsonObject schema = grabSchema(MAP_CLICK);
+    List<Field> fields = grabClassFields(MapClickEvent.class);
+
+    schemaContainsFields(schema, fields);
+  }
+
+  @Test
+  public void checkMapDragEndEventSize() throws Exception {
+    JsonObject schema = grabSchema(MAP_DRAG);
+    List<Field> fields = grabClassFields(MapDragendEvent.class);
+
+    assertEquals(schema.size(), fields.size());
+  }
+
+  @Test
+  public void checkMapDragEndEventFields() throws Exception {
+    JsonObject schema = grabSchema(MAP_DRAG);
+    List<Field> fields = grabClassFields(MapDragendEvent.class);
+
+    schemaContainsFields(schema, fields);
+  }
+
+  @Test
+  public void checkMapLoadEventSize() throws Exception {
+    JsonObject schema = grabSchema(MAP_LOAD);
+    List<Field> fields = grabClassFields(MapLoadEvent.class);
+
+    //FIXME: this assertion is invalid: we should introduce a concept of mandatory/optional field to schema validation
+    //assertEquals(schema.size(), fields.size());
+  }
+
+  @Test
+  public void checkMapLoadEventFields() throws Exception {
+    JsonObject schema = grabSchema(MAP_LOAD);
+    List<Field> fields = grabClassFields(MapLoadEvent.class);
+
+    schemaContainsFields(schema, fields);
+  }
+
```

Comment:
Can we expand SchemaTest.java to support all events we have in https://github.com/mapbox/mapbox-gl-native/tree/37d17dc2b986c587f06499265f34b49b8bc78f6f/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry, not all are being tested atm.